oci/layout/oci_dest.go

```diff
@@ -32,7 +33,16 @@ func newImageDestination(ref ociReference) (types.ImageDestination, error) {
 			SchemaVersion: 2,
 		},
 	}
-	return &ociImageDestination{ref: ref, index: index}, nil
+	d := &ociImageDestination{ref: ref, index: index, sharedBlobDir: ctx.OCISharedBlobDirPath}
```

Comment:
`ctx` may be `nil`; it might be easier to pass the `*types.SystemContext` all the way through to `blobPath`.